src/status.rs

```diff
@@ -85,12 +88,14 @@ impl StatusCode {
         let b = src[1].wrapping_sub(b'0') as u16;
         let c = src[2].wrapping_sub(b'0') as u16;
 
-        if a == 0 || a > 5 || b > 9 || c > 9 {
+        if a == 0 || a > 9 || b > 9 || c > 9 {
             return Err(InvalidStatusCode::new());
         }
 
         let status = (a * 100) + (b * 10) + c;
-        Ok(StatusCode(status))
+        NonZeroU16::new(status)
+            .map(StatusCode)
+            .ok_or_else(InvalidStatusCode::new)
```

Comment:
Looks like the allows (now as well as before) status codes < 100, should probably just be

```suggestion
        StatusCode::from_u16(status)
```

Reply:
see L91

Reply:
Oh, I missed the `a == 0` part. Nevermind.